src/server/sass/build.js

```diff
@@ -69,19 +91,97 @@ export class Build {
       sourceMap: true,
       sourceMapEmbed: true,
       includePaths: [
-        path.resolve(__dirname, '../..'),
-        path.resolve(__dirname, '../../../node_modules'),
+        resolve(__dirname, '../..'),
+        resolve(__dirname, '../../../node_modules'),
       ],
       importer: this.theme === 'dark' ? DARK_THEME_IMPORTER : undefined
     });
 
-    const prefixed = postcss([ autoprefixer ]).process(rendered.css);
+    const processor = postcss([ autoprefixer ]);
+
+    const urlAssets = [];
+
+    if (this.urlImports) {
+      processor.use(postcssUrl({
+        url: (request) => {
+          if (!request.pathname) {
+            return request.url;
+          }
+
+          const asset = makeAsset(request, (
+            request.pathname.startsWith('ui/assets')
+              ? {
+                path: resolve(UI_ASSETS_DIR, relative('ui/assets', request.pathname)),
+                root: UI_ASSETS_DIR,
+                boundry: UI_ASSETS_DIR,
+                urlRoot: `ui/`,
+              }
+              : {
+                path: resolve(this.sourceDir, request.pathname),
+                root: this.sourceDir,
+                boundry: this.urlImports.publicDir,
+                urlRoot: this.urlImports.urlBase,
+                copyRoot: this.targetDir,
+              }
+          ));
+
+          if (!urlAssets.some(({ path, copyTo }) => path === asset.path && copyTo === asset.copyTo)) {
```

Comment:
Does this run per-plugin, or globally? If globally, might be worth refactoring this into a map or set lookup to see if this asset has already been processed.

Reply:
it happens per entry file, but unless we have tens of thousands of url calls then it shouldn't have any perceptible performance cost.